core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2582,13 +2582,21 @@ public void resetFile(long fileId)
 
   @Override
   public void setAttribute(AlluxioURI path, SetAttributeOptions options)
-      throws FileDoesNotExistException, AccessControlException, InvalidPathException {
+      throws FileDoesNotExistException, AccessControlException, InvalidPathException,
+      IOException {
     Metrics.SET_ATTRIBUTE_OPS.inc();
     // for chown
     boolean rootRequired = options.getOwner() != null;
     // for chgrp, chmod
     boolean ownerRequired =
         (options.getGroup() != null) || (options.getMode() != Constants.INVALID_MODE);
+    if (options.getOwner() != null && options.getGroup() != null) {
+      try {
+        checkUserBelongsToGroup(options.getOwner(), options.getGroup());
+      } catch (IOException e) {
+        throw new IOException("Could not setOwner for " + path.toString() + ".", e);
```

Comment:
We need to re-throw it with updated error message because the message contained in the original IOException is general. It just tells the upper caller that either a) user does not belong to group, or b) the user does not exist. It does not tell the end user that trying to chown is the operation that caused this error. Thus we decided that we catch the original exception and do a little interpretation of its meaning. I will think about this.